encoding/string/string.go

```diff
@@ -0,0 +1,29 @@
+package string
+
+import "bytes"
+
+type encoder struct {
+	buf *bytes.Buffer
+}
+
+// NewEncoder returns a new encoder instance.
+func NewEncoder() Encoder {
+	return &encoder{
+		buf: bytes.NewBuffer(make([]byte, 0)),
```

Comment:
Any reason not to just do
```
type encoder struct {
  buf bytes.Buffer
}
```
?

Reply:
I had considered making the initial buffer size configurable and `bytes.NewBuffer` returns a `*bytes.Buffer` but I'll just use `bytes.Buffer` for now as it doesn't require initialization.

Reply:
In that case it's fine to leave it as is, didn't know that was your intention based on the PR.

For that purpose I'd imagine you'd pass in an options for specifying the initial buffer sizes etc.